ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/cache/CachedWorldStorageManager.java

```diff
@@ -198,4 +211,66 @@ public void onClearTrieLog() {
   public void onCloseStorage() {
     this.cachedWorldStatesByHash.clear();
   }
+
+  @VisibleForTesting
+  @Override
+  protected TrieLogFactory setupTrieLogFactory(final BesuContext pluginContext) {
+    // if we have a TrieLogService from pluginContext, use it.
+
+    var trieLogService =
+        Optional.ofNullable(pluginContext)
+            .flatMap(context -> context.getService(TrieLogService.class));
+
+    if (trieLogService.isPresent()) {
+      trieLogService.get().configureTrieLogProvider(getTrieLogProvider());
+      return trieLogService.get().getTrieLogFactory();
+    } else {
+      // Otherwise default to TrieLogFactoryImpl
+      return new TrieLogFactoryImpl();
+    }
+  }
+
+  @VisibleForTesting
+  TrieLogProvider getTrieLogProvider() {
+    return new TrieLogProvider() {
+      @Override
+      public <T extends TrieLog.LogTuple<?>> Optional<TrieLog> getTrieLogLayer(
+          final Hash blockHash) {
+        return CachedWorldStorageManager.this.getTrieLogLayer(blockHash);
+      }
+
+      @Override
+      public <T extends TrieLog.LogTuple<?>> Optional<TrieLog> getTrieLogLayer(
+          final long blockNumber) {
+        return CachedWorldStorageManager.this
+            .blockchain
+            .getBlockHeader(blockNumber)
+            .map(BlockHeader::getHash)
+            .flatMap(CachedWorldStorageManager.this::getTrieLogLayer);
+      }
+
+      @Override
+      public <T extends TrieLog.LogTuple<?>> List<TrieLog> getTrieLogsByRange(
+          final long fromBlockNumber, final long toBlockNumber) {
+        return rangeAsStream(fromBlockNumber, toBlockNumber)
+            .map(blockchain::getBlockHeader)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .map(BlockHeader::getHash)
+            .map(CachedWorldStorageManager.this::getTrieLogLayer)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(Collectors.toList());
+      }
+
+      Stream<Long> rangeAsStream(final long fromBlockNumber, final long toBlockNumber) {
+        if (Math.abs(toBlockNumber - fromBlockNumber) > 1000L) {
```

Comment:
Where is this limit number coming from?

Reply:
Since there isn't a use for this feature other than the shomei plugin currently, 1k is a compromise between shomei batch sync sizes and a reasonable upper bound on an corresponding rpc payload size.  The payload size can vary based on the chain's gas limit and operations performed during the range.  Using linea as a guidepost, an rpc request for blocks 0-1k weighs in at 510kb, whereas blocks 600k->601k are ~60mb.

I will add a class level constant for this rather than a 'magic number' however.  :+1: